lightning/src/ln/onchain_utils.rs

```diff
@@ -0,0 +1,946 @@
+			},
+			PackageTemplate::CounterpartyHTLCTx { ref inputs } => {
+				for outp in inputs.keys() {
+					bumped_tx.input.push(TxIn {
+						previous_output: *outp,
+						script_sig: Script::new(),
+						sequence: 0xfffffffd,
+						witness: Vec::new(),
+					});
+				}
+				for (i, (outp, rem)) in inputs.iter().enumerate() {
+					log_trace!(logger, "Claiming outpoint {}:{}", outp.txid, outp.vout);
+					if let Ok(chan_keys) = TxCreationKeys::derive_new(&onchain_handler.secp_ctx, &rem.per_commitment_point, &rem.counterparty_delayed_payment_base_key, &rem.counterparty_htlc_base_key, &onchain_handler.key_storage.pubkeys().revocation_basepoint, &onchain_handler.key_storage.pubkeys().htlc_basepoint) {
+						let witness_script = chan_utils::get_htlc_redeemscript_with_explicit_keys(&rem.htlc, &chan_keys.broadcaster_htlc_key, &chan_keys.countersignatory_htlc_key, &chan_keys.revocation_key);
+
+						if !rem.preimage.is_some() { bumped_tx.lock_time = rem.htlc.cltv_expiry }; // Right now we don't aggregate time-locked transaction, if we do we should set lock_time before to avoid breaking hash computation
+						if let Ok(sig) = onchain_handler.key_storage.sign_counterparty_htlc_transaction(&bumped_tx, i, &rem.htlc.amount_msat / 1000, &rem.per_commitment_point, &rem.htlc, &onchain_handler.secp_ctx) {
+							bumped_tx.input[i].witness.push(sig.serialize_der().to_vec());
+							bumped_tx.input[i].witness[0].push(SigHashType::All as u8);
+							if let Some(preimage) = rem.preimage {
+								bumped_tx.input[i].witness.push(preimage.0.to_vec());
+							} else {
+								// Due to BIP146 (MINIMALIF) this must be a zero-length element to relay.
+								bumped_tx.input[i].witness.push(vec![]);
+							}
+							bumped_tx.input[i].witness.push(witness_script.clone().into_bytes());
+						}
+					}
+				}
+				log_trace!(logger, "Going to broadcast Claim Transaction {} claiming counterparty htlc output...", bumped_tx.txid());
+				return Some(bumped_tx);
+			},
+			PackageTemplate::HolderHTLCTx { ref input } => {
+				let htlc_tx = onchain_handler.get_fully_signed_htlc_tx(&input.0, &input.1.preimage);
+				if let Some(htlc_tx) = htlc_tx {
+					// Timer set to $NEVER given we can't bump tx without anchor outputs
+					log_trace!(logger, "Going to broadcast Holder HTLC-{} claiming HTLC output {} from {}...", if input.1.preimage.is_some() { "Success" } else { "Timeout" }, input.0.vout, input.0.txid);
+					return Some(htlc_tx);
+				}
+				return None;
+			},
+			PackageTemplate::HolderCommitmentTx { ref input, .. } => {
+				let signed_tx = onchain_handler.get_fully_signed_holder_tx(&input.1.funding_redeemscript).unwrap();
+				// Timer set to $NEVER given we can't bump tx without anchor outputs
+				log_trace!(logger, "Going to broadcast Holder Transaction {} claiming funding output {} from {}...", signed_tx.txid(), input.0.vout, input.0.txid);
+				return Some(signed_tx);
+			}
+		}
+	}
+	pub(crate) fn package_cpfp<U: Deref>(&mut self, utxo_pool: &U)
```

Comment:
perhaps rename this to `maybe_bump_package_via_cpfp`, remove the panic, and move the conditional on the strategy from the caller to here?